File: shop-frontend/src/app/model/class/Product.ts
export class Product {
  productId: number;
  productCode: string;
  productName: string;
  productCategory: string;
  unitPrice: number;
  stock: number;
  reorderPoint: number;

  constructor() {
    this.productId = 0;
    this.productCode = '';
    this.productName = '';
    this.productCategory = '';
    this.unitPrice = 0;
    this.stock = 0;
    this.reorderPoint = 0;
  }
}

File: shop-backend/Models/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace shop_backend.Models
{
    public class Product
    {
        public int ProductId { get; set; } // Primary Key

        public string ProductCode { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public string ProductCategory { get; set ;} = string.Empty;
        public float UnitPrice {get; set;}
        public int Stock { get; set; }
        public int ReorderPoint { get; set; }
    }
}